src/jdk.incubator.vector/share/classes/jdk/incubator/vector/CPUFeatures.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.incubator.vector;
+
+import jdk.internal.vm.vector.VectorSupport;
+
+import java.util.Set;
+
+import static jdk.incubator.vector.Util.requires;
+import static jdk.internal.util.Architecture.isX64;
+import static jdk.internal.vm.vector.Utils.debug;
+
+/**
+ * Enumerates CPU ISA extensions supported by the JVM on the current hardware.
+ */
+/*package-private*/ class CPUFeatures {
+    private static final Set<String> features = getCPUFeatures();
+
+    private static Set<String> getCPUFeatures() {
+        String featuresString = VectorSupport.getCPUFeatures();
+        debug(featuresString);
+        String[] features = featuresString.toLowerCase().split(", "); // ", " is used as a delimiter
```

Comment:
Please use `toLowerCase(Locale.ROOT)`: if the system locale is turkish, `I` and dotless i are two letters, and the dotless i will fail in the subsequent `validateFeatures` assertion. Same for `hasFeature`.

Reply:
Good point. Fixed.